src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Attribute;
+
+use Symfony\Component\Console\Completion\CompletionInput;
+use Symfony\Component\Console\Completion\Suggestion;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class Argument
+{
+    private ?int $mode = null;
+
+    /**
+     * Represents a console command <argument> definition.
+     *
+     * If unset, the `name`, `mode` and `default` values might be inferred from the parameter definition.
+     *
+     * @param string|bool|int|float|array|null                               $default         The default value (for InputArgument::OPTIONAL mode only)
+     * @param array|callable-string(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
+     */
+    public function __construct(
+        public string $name = '',
+        public string $description = '',
+        public string|bool|int|float|array|null $default = null,
+        public array|string $suggestedValues = [],
+    ) {
+    }
+
+    public static function tryFrom(\ReflectionParameter $parameter): ?self
+    {
+        /** @var self $self */
+        if (null === $self = ($parameter->getAttributes(self::class, \ReflectionAttribute::IS_INSTANCEOF)[0] ?? null)?->newInstance()) {
+            return null;
+        }
+
+        $type = $parameter->getType();
+        $name = $parameter->getName();
+
+        if (!$type instanceof \ReflectionNamedType) {
+            throw new LogicException(\sprintf('The parameter "$%s" must have a named type. Untyped, Union or Intersection types are not supported for command arguments.', $name));
+        }
+
+        $parameterTypeName = $type->getName();
+
+        if (!\in_array($parameterTypeName, ['string', 'bool', 'int', 'float', 'array'], true)) {
+            throw new LogicException(\sprintf('The type "%s" of parameter "$%s" is not supported as a command argument. Only "string", "bool", "int", "float", and "array" are allowed.', $parameterTypeName, $name));
+        }
+
+        if (!$self->name) {
+            $self->name = $name;
+        }
+
+        $self->mode = null !== $self->default || $parameter->isDefaultValueAvailable() ? InputArgument::OPTIONAL : InputArgument::REQUIRED;
+        if ('array' === $parameterTypeName) {
+            $self->mode |= InputArgument::IS_ARRAY;
+        }
```

Comment:
For later, we set `InputArgument::IS_ARRAY` for variadic arguments (provided the type is not `array`). This would work because a command can only have one `InputArgument` with the `IS_ARRAY` mode. For `InputOption`, this would be limited to a single option or argument.